dependencies/scripts/download_packages.py

```diff
@@ -25,16 +26,35 @@ def parse_pypi(data: dict[str, Any]) -> list[str]:
     return [row["project"] for row in data["rows"]]
 
 
-ECOSYSTEMS = {
-    "npm": {
-        "url": "https://packages.ecosyste.ms/api/v1/registries/npmjs.org/packages?per_page=10000&page=1&sort=downloads",
-        "parser": parse_npm,
-    },
-    "pypi": {
-        "url": "https://hugovk.github.io/top-pypi-packages/top-pypi-packages.min.json",
-        "parser": parse_pypi,
-    },
-}
+class ServerError(Exception):
+    """Custom exception for HTTP 5xx errors."""
+
+
+@dataclass
```

Comment:
```suggestion
@dataclass(frozen=True)
```

Maybe? These are read-only no?